egui/src/grid.rs

```diff
@@ -263,14 +385,40 @@ impl Grid {
         Self {
             id_source: Id::new(id_source),
             striped: false,
+            header_row: false,
             min_col_width: None,
             min_row_height: None,
             max_cell_size: Vec2::INFINITY,
             spacing: None,
+            color_spec: Default::default(),
             start_row: 0,
         }
     }
 
+    /// Replace the color picker for columns
+    pub fn with_column_color(mut self, predicate: ColorPickerFn) -> Self {
+        self.color_spec.col_pickers = vec![predicate];
+        self
+    }
+
+    /// Add a `ColorPickerFn` to the color spec without overwriting existing column colors.
+    pub fn add_column_color(mut self, predicate: ColorPickerFn) -> Self {
+        self.color_spec.col_pickers.push(predicate);
+        self
+    }
+
+    /// Replace the color picker for rows
+    pub fn with_row_color(mut self, predicate: ColorPickerFn) -> Self {
```

Comment:
Do we really need both `add` and `with`? Seems like we can do without the `with` (why would anyone add colors and then replace them later?)

Reply:
👍